controller/src/main/java/com/controller/Main.java

```diff
@@ -21,28 +23,34 @@
     public static void main(String[] args) {
         // Parse command line argument
         if(args.length % 2 != 0) {
-            System.out.println("Command line argument mal formed");
-            return;
+            throw new MalformedParametersException("Command line argument is malformed");
         }
         int time = DEFAULT_TIME; // set time to default
+        String configFileName = "input_config.json"; //default config file name
         for(int i = 0; i < args.length; i += 2){
             String flag = args[i];
             String val = args[++i];
             switch (flag) {
                 case "-t" :
                     time = Integer.valueOf(val);
+                    if(time < 0) {
+                        System.out.println("Invalid time would be ignored");
+                        time = DEFAULT_TIME;
+                    }
                     break;
-                default:
-                    System.out.println("Invalid flag: " + flag);
+                case "-f" :
+                    configFileName = val;
                     break;
+                default:
+                    throw new MalformedParametersException("invalid flag");
             }
         }
 
         // Parse input config file
-        String configFileName = "input_config.json";
         HashMap<String, String> input = ConfigFileParser.getInputFromConfigFile(configFileName);
```

Comment:
Do we have a schema validation file that we can use to validate the user's input_config.json?

Reply:
Just added.